planner/core/rule_partition_processor.go

```diff
@@ -472,6 +478,33 @@ func partitionRangeForOrExpr(sctx sessionctx.Context, expr1, expr2 expression.Ex
 	return tmp1.union(tmp2)
 }
 
+func partitionRangeForInExpr(sctx sessionctx.Context, args []expression.Expression,
+	pruner *rangePruner) partitionRangeOR {
+	if len(args) <= 1 {
+		return pruner.fullRange()
+	}
+	col, ok := args[0].(*expression.Column)
+	if !ok || col.ID != pruner.col.ID {
```

Comment:
when will `col.ID != pruner.col.ID` happen?

Reply:
Return `fullRange`, means do not prune:

```
partition expr = by range (a)
query = b in (xxx)
```

It's obvious for this case.

Reply:
ok

we need a test case for it

Reply:
done